lib/receivers/httpReceiver.js

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2016 Kinvey Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const log = require('../log/logger');
+
+module.exports = (() => {
+  let server = null;
+  let app = null;
+
+  function healthCheck(req, res, next) {
+    return res.status(200).json({ healthy: true });
+  }
+
+  function generateBaseTask(req, res, next) {
+    let appMetadata;
+
+    try {
+      appMetadata = JSON.parse(req.get('X-Kinvey-App-Metadata'));
+    } catch (e) {
+      appMetadata = {};
+    }
+
+    req.task = {
+      appId: req.get('X-Kinvey-Environment-Id'),
+      appMetadata,
+      authKey: req.get('X-Auth-Key'),
+      requestId: req.get('X-Kinvey-Request-Id'),
+      method: req.method.toUpperCase(),
+      taskType: 'dataLink',
+      request: {
+        method: req.method,
+        headers: req.get('X-Kinvey-Original-Request-Headers'),
+        username: req.get('X-Kinvey-Username'),
+        userId: req.get('X-Kinvey-User-Id'),
+        serviceObjectName: req.params.serviceObjectName,
+      },
+      response: {
+        status: req.get('X-Kinvey-Response-Status') || 0,
+        headers: req.get('X-Kinvey-Response-Headers') || {},
+        body: req.get('X-Kinvey-Response-Body') || {}
```

Comment:
They may be part of the request.  `undefined/null` checks aren't necessary because  those headers will eval to `null` if they aren't there, and the `||` will set the value in the `response` object to the default (`0` or `{}`, depending...)